tests/csapi/ignored_users_test.go

```diff
@@ -0,0 +1,105 @@
+// +build !dendrite_blacklist
+
+// Rationale for being included in Dendrite's blacklist: https://github.com/matrix-org/dendrite/issues/600
+package csapi_tests
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+// The Spec says here
+//     https://spec.matrix.org/v1.1/client-server-api/#server-behaviour-13
+// that
+// > Servers must not send room invites from ignored users to clients.
+//
+// Synapse does not have this property, as detailed in
+// https://github.com/matrix-org/synapse/issues/11506.
+// This reproduces that bug.
+func TestInviteFromIgnoredUsersDoesNotAppearInSync(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintCleanHS)
+	defer deployment.Destroy(t)
+	alice := deployment.RegisterUser(t, "hs1", "alice", "sufficiently_long_password_alice")
+	bob := deployment.RegisterUser(t, "hs1", "bob", "sufficiently_long_password_bob")
+	chris := deployment.RegisterUser(t, "hs1", "chris", "sufficiently_long_password_chris")
+
+	// Alice creates a room for herself.
+	public_room := alice.CreateRoom(t, map[string]interface{}{
+		"preset": "public_chat",
+	})
+
+	// Alice waits to see the join event.
+	alice.SyncUntilTimelineHas(
+		t, public_room, func(ev gjson.Result) bool {
+			return ev.Get("type").Str == "m.room.member" &&
+				ev.Get("state_key").Str == alice.UserID &&
+				ev.Get("content.membership").Str == "join"
+		},
+	)
+
+	// Alice ignores Bob.
+	alice.MustDoFunc(
+		t,
+		"PUT",
+		[]string{"_matrix", "client", "v3", "user", alice.UserID, "account_data", "m.ignored_user_list"},
```

Comment:
Dendrite doesn't implement any `/v3/` endpoint. Is there some spec documentation on the changes between `r0` and `v3`?

Reply:
> Dendrite doesn't implement any `/v3/` endpoint. Is there some spec documentation on the changes between `r0` and `v3`?

All r0 endpoints became v3 with Matrix 1.1 -- @DMRobertson Maybe we should just use r0 for now here?

Reply:
Happy to use r0; I was just blindly going by what the latest spec said.